### ISSUE TITLE
API Break - QgsVectorLayer

### ISSUE THREAD

**Issue by merkato** [blazek/lrs#18]
QGIS: 2.99.0-Master Master, 6c64c5a 
LRS v 1.0.1

AttributeError: 'QgsVectorLayer' object has no attribute 'pendingFields' 
Traceback (most recent call last):
  File "/home/mechanik/.local/share/QGIS/QGIS3/profiles/default/python/plugins/lrs/ui/lrsfieldcombomanager.py", line 73, in layerChanged
    self.resetFields()
  File "/home/mechanik/.local/share/QGIS/QGIS3/profiles/default/python/plugins/lrs/ui/lrsfieldcombomanager.py", line 99, in resetFields
    for idx, field in enumerate(layer.pendingFields()):
AttributeError: 'QgsVectorLayer' object has no attribute 'pendingFields'

**Comment by blazek** [comment]
Another API break Jan 15 2018 https://github.com/qgis/QGIS/pull/6050

I'l fix that after 3.0 release.

**Comment by merkato** [comment]
#19

**Comment by merkato** [comment]
```
 File ..python/plugins/lrs/lrs/lrscalib.py, line 79, in __init__
			    self.distanceArea.setSourceCrs(self.crs)
			TypeError: QgsDistanceArea.setSourceCrs(): not enough arguments
```

tried: 
`        self.distanceArea.setSourceCrs(self.crs,QgsProject.instance().transformContext())`

doesn't help, same error

**Comment by blazek** [comment]
Reopened, self.distanceArea.setSourceCrs not yet fixed.

**Comment by blazek** [comment]
> self.distanceArea.setSourceCrs(self.crs,QgsProject.instance().transformContext())

For me it works, maybe you have not reloaded plugin?

I have pushed this one plus other pendingFields() occurences.

**Comment by merkato** [comment]
Indeed. QGIS doesn't closed properly. After clean restart, everything is OK

**Comment by blazek** [comment]
Released 1.0.2
